sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -299,6 +297,9 @@ case class SampleExec(
           | }
          """.stripMargin.trim)
 
+      ctx.addMutableState(s"$samplerClass<UnsafeRow>", sampler,
```

Comment:
This change should stay the way it is. Notice that the initialization code for the `addMutableState` call on the line just below is the same code passed to the `addNewFunction` call. This means that when we create the new function, we may get back a class-qualified function name (which here we store in `initSamplerFuncName`), so the call to `addMutableState` must come after the `addNewFunction` call.